### DISCUSSION TITLE
Providing images via getStaticProps

### DISCUSSION THREAD

**Discussion by Laityned** [vercel/next.js#41548] [Help]
```typescript
export const getStaticProps: GetStaticProps<
  ChannelPageProps,
  ChannelPageParams
> = async ({ params }) => {
  const serializedChannels = channelPages.map(serializeChannelPage)
  const channel = serializedChannels.find(({ slug }) => slug === params!.slug)!

  return {
    props: channel,
  }
}

let preventDeathcode = channelPages
```

I have a staticprops function which should provided the page an object containing a StaticImageData (in this case the backgroundImage inside the channelpage). However as described in issue #29571 the dead-code elimination prevents that the images is placed in the static/media folder.  An option to prevent this dead-code is by just randomly assigning a variable like preventDeathcode. What would you recommended as the correct approach here?

**Comment by icyJoseph** [comment]
Perhaps something like, https://next-code-elimination.vercel.app/s/3LSEwuipR

But that applies to #29571 though. Not sure what's `serializeChannelPage` doing, and if there's any way you could abstract it away, you could do this https://next-code-elimination.vercel.app/s/ei-cgsRr_

```js

import Head from 'next/head'
import Image from 'next/image'
import styles from '../styles/Home.module.css'
import directImportImage from '../public/robert-bye-Fr1t7UHYkJU-unsplash.jpg'
import staticPropImage from '../public/robert-bye--P2SwAbYcbs-unsplash.jpg'

const channelPages = [directImportImage, staticPropImage]

const findChannel = (channels, params) => channels.map(serializeChannelPage).find(({slug}) => slug === params.slug)

export const getStaticProps: GetStaticProps<
  ChannelPageProps,
  ChannelPageParams
> = async ({ params }) => {
  const channel = findChannel(channelPages, params)

  return {
    props: {channel, params}
  }
}

export default function Home({ params, channel = findChannel(channelPages, params) }) {
  return (
    <div className={styles.container}>
      <main className={styles.main}>
      </main>
    </div>
  )
}
```

The key part would be the default parameter magic done here:

```js
const serializeChannelPage = (x) => x 
const findChannel = (channels, params) => channels.map(serializeChannelPage).find(({slug}) => slug === params.slug)
const inputChannels = [{slug: 'abc'}]
// if channel is undefined or not passed, then the findChannel fn is called
// null or any other value does not trigger findChannel
const log = ({params, channel = findChannel(inputChannels, params)}) => console.log({params, channel})

log({params: {slug: 'abc'}})
```

**Reply by Laityned** [reply]
To summarize, instead of writing a seperate variable that will prevent dead-code, you will pass it as an default parameter to the page.
However, this still means that the variable channelPages (which include data of all static pages) is passed to the client bundle.

Concluding that both approaches will have the same client bundle as solution.

the data of the staticImageData is passed correctly (src, width, height and so on), however nextjs does not make the image available at _next/static/media/.... is there not a solution to that?

Or should this be considered a bug in nextjs that static importing images on server side will not make the image available at the static media path of the server

**Reply by Laityned** [reply]
see #41554 for possible solution
